File: src/kbs/baekjoon/silver/Baekjoon2579.java
package kbs.baekjoon.silver;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Baekjoon2579 {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int T=Integer.parseInt(br.readLine());
		int stairs[]=new int[T+1];
		int dp[]=new int[T+1];
		for(int i=0;i<T;i++) {
			stairs[i]=Integer.parseInt(br.readLine());
		}
		dp[0]=stairs[0]; dp[1]=stairs[0]+stairs[1];
		if(T>=2) {	
		dp[2]= Math.max(stairs[1]+stairs[2], stairs[0]+stairs[2]);
			for(int i=3;i<T;i++) {
				dp[i]=Math.max(dp[i-2]+stairs[i], dp[i-3]+stairs[i-1]+stairs[i]);
			}
		}
		System.out.println(dp[T-1]);
	}

}

File: src/ljh/baekjoon/silver/baekjoon_2579.py
# baekjoon 알고리즘 2579
# 2023.09.23 이주현

import sys

num = int(sys.stdin.readline())
stairs = [0]*301
for i in range(num):
    stairs[i]=int(input())

DP = [0]*301
DP[0] = stairs[0] # 첫번째 계단
DP[1] = stairs[0]+stairs[1] # 두번째 계단
DP[2] = max(stairs[0]+stairs[2], stairs[1]+stairs[2]) # 세번째 계단

for i in range(3,num):
    DP[i] = max(DP[i-3] + stairs[i-1] + stairs[i], DP[i-2]+stairs[i])

print(DP[num-1])